system/lib/sbrk.c

```diff
@@ -6,13 +6,21 @@
  *
 */
 
+#if __EMSCRIPTEN_PTHREADS__
+#define RETRY_SBRK 1
+#endif
+
+#ifdef __EMSCRIPTEN_WASM_WORKERS__
+#define RETRY_SBRK 1
```

Comment:
Combining these into a single block makes sense to me. 

Also, perhaps instead of `defined(__EMSCRIPTEN_PTHREADS__) || defined(__EMSCRIPTEN_WASM_WORKERS__)` you can just use `ifdef __EMSCRIPTEN_SHARED_MEMORY__` ?  Or equivalently `ifdef _REENDTRANT`

Reply:
Hmm well, I suppose this could now just read `#ifdef __EMSCRIPTEN_SHARED_MEMORY__` everywhere without introducing a `RETRY_SBRK` define at all?

Reply:
Ah, I had thought to keep it for the brevity (since __EMSCRIPTEN_SHARED_MEMORY__ is kind of long), and maybe for future use. Anyway, changed to ```#ifdef __EMSCRIPTEN_SHARED_MEMORY__ ``` as you suggested.